falcon/api_helpers.py

```diff
@@ -167,3 +169,32 @@ def new_fn(req, resp, exception):
             resp.content_type = media_type
 
     return new_fn
+
+
+class CloseableStreamIterator(six.Iterator):
+    """
```

Comment:
Nitpick: Would you mind reformatting this according to [PEP-257](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings)?

> Multi-line docstrings consist of a summary line just like a one-line docstring, followed by a blank line, followed by a more elaborate description.

E.g.:

```py
class CloseableStreamIterator(six.Iterator):
    """Short summary line ending in a period.

    Detailed description.
    """
```

Also, it may be helpful to mention why a "closeable" iterator is necessary.